src/expr.cpp

```diff
@@ -2647,6 +2648,53 @@ void BinaryExpr::Print() const {
     pos.Print();
 }
 
+std::pair<llvm::Constant *, bool> BinaryExpr::GetConstant(const Type *type) const {
```

Comment:
This function vigorously needs comments explaining isNotValidForGlobal logic and the concept as a whole.